### PR TITLE
Add `org_name` config param

### PR DESCRIPTION
**Pull request by outcatcher**
Add `org_name` as alternative to `ins_id`

Part of #53

### CODE REVIEW

**Comment by martinbaillie** [comment]
Thanks very much! Looks great. I'll try to get it properly reviewed and released in the coming days.

File: github/client.go [hunk 1]
```diff
@@ -177,6 +188,72 @@ func (c *Client) Token(ctx context.Context, opts *tokenOptions) (*logical.Respon
 	return tokenRes, nil
 }
 
+func (c *Client) getInstallationID(config *Config) (int, error) {
```
Comment (martinbaillie) [inline github/client.go:191]: Could you please write tests for this new method? Thanks.

Comment (lego963) [reply github/client.go:192]: I will notify you when I will finish

Comment (lego963) [reply github/client.go:193]: @martinbaillie can you review again please?

**Review by martinbaillie — APPROVED** [review]

**Comment by martinbaillie** [comment]
Thanks again! This is in the https://github.com/martinbaillie/vault-plugin-secrets-github/releases/tag/v1.3.0.

**Comment by grahamc** [comment]
I'm a bit afraid this PR makes it a bit harder to move to multiple installations in a single mount, since it makes the top level configuration more complicated and creates an ID discovery phase tied to the overall configuration.

It seems to me that moving to multiple installations per mount is going to imply removing support for a single installation ID / app name being configured on the mount, and pushing it in to the requests, similar to how `repository_ids` is passed.

I worry that having the plugin support both use cases may make for more complex code in a security-sensitive path.

To that end I'd like to ask a bit directly of @martinbaillie -- is multiple installations something you want to support, and if so, do you have an idea of how it would look?

**Comment by martinbaillie** [comment]
Apologies, I've been away for a bit. Thanks very much for your thought on this issue.

I first wrote this plugin with my previous gig in mind. They were a _one-install-one-mount shop_, and without much thought, I assumed other places would use multiple mounts if they had more installations. 

However, I've heard now from a few companies and it is apparent that the additional flexibility on a single mount is desirable. It is indeed now possible (and even implemented #63 🎉) without additional cognitive overhead on the end user (i.e. "what `ins_id` do I use for this request?") thanks to the recent Permissions Sets feature. Previously to keep those details abstracted from the user necessitated multiple mounts.

With that in mind, I think supporting the multiple installations on a single mount _is the right direction_ for this plugin to head. It is also more aligned with the GitHup App model of `1-app:N-installs`.

I'm going to look to merge @grahamc's #63 and work on cutting probably a new major release as this is an API break. 

As for the `org_name` work introduced in this PR, I would be keen to learn more about how you were using the plugin @outcatcher / @lego963 to see if we could still accommodate tokens by org name. I should have asked more before hastily merging last month—sorry about that.

**Comment by outcatcher** [comment]
It seems pretty strange for us to use installation ID every time, needing to store it instead of clearly known org name. We were thinking about caching those IDs, but we wanted to get a basic functionality first. Anyway, we already have a different implementation of #63 feature in our fork, but we'll backport caching `ins_id` if we implement it in the nearest future.

**Comment by martinbaillie** [comment]
It was discussed a bit more here: https://github.com/martinbaillie/vault-plugin-secrets-github/issues/53#issuecomment-1030920001

I'll re-instate some form of `ins_id` lookup to both /token and permission sets for 2.0.0.

**Pull request merged**